src/u-blox_GNSS.cpp

```diff
@@ -1510,7 +1516,7 @@ void DevUBLOXGNSS::process(uint8_t incoming, ubxPacket *incomingUBX, uint8_t req
 
     // If user has assigned an output port then pipe the characters there,
     // but only if the port is different (otherwise we'll output each character twice!)
-    if (_outputPort._serialPort != _ubxOutputPort._serialPort)
+    if (_outputPort != _ubxOutputPort)
```

Comment:
Hi Michael (@mazgch ),

I'm puzzled by this change (and the next three changes). The code is supposed to check if _outputPort is using a different Stream / port to _ubxOutputPort. This is to prevent the character being output twice if the Stream is the same.

If I understand correctly, _outputPort will always be != to _ubxOutputPort. They are different instances at different addresses. I believe the code should now be:

```
if (_outputPort._outputPort != _ubxOutputPort._outputPort)
```

I think you will also need to make _outputPort public in sfe_bus.h. If it is private, the code in process cannot access it. I tried to use friend, but I could not get it to work.

Thank you,
Paul

Reply:
I have overridden the == and != operator of the SfePrint class this means if you call equal on the object itself it in fact calls the operator and compares its private _outputPort member of the two objects. The change is not really needed but I read you comment in the SfeStream class and found you might find this approach useful. 

Does it not work for you as expected, or dou you need the access to its internal member for something else?

Reply:
You could also add a 

bool Sfeprint::equals(SfePrint &rhs) const 
{ 
  return _outputPort == rhs._outputPort; 
} 

instead of overloading the == operator and use this like

if (!_outputPort.equals(_ubxOutputPort))

The the _outputPort can be protected or private without makeing the other class a friend.

Reply:
Ah! Very good. I had missed [these two lines](https://github.com/sparkfun/SparkFun_u-blox_GNSS_v3/pull/3/files#diff-ee49d6d71c5c4c44f48a1d36fe6d87cbb5d34bafb3092995d83c3c312cf660e4R198-R199).

Resolved!